### DISCUSSION TITLE
Switch iModel through backstage

### DISCUSSION THREAD

**Discussion by kmurari103** [iTwin/itwinjs-core#2976] [Q&A]
Hi there,
I'm working on a web viewer app and able to switch to different iModel by changing `iModelId `& `contextId ` in URL but i want to do it from backstage. Here's snippet for that- 
![image](https://user-images.githubusercontent.com/27068008/148054507-5edd72f5-9fd9-4b6e-90b1-e066bc9efe0e.png)

For this I used `bentley/imodel-select-react` package (able to get desired functionality) but it seems to be deprecated and suggestion is to use `@itwin/imodel-browser-react` which is not stable as per the [documentation ](https://www.npmjs.com/package/@itwin/imodel-browser-react#:~:text=Consider%20this%20package%20as%20unstable%20until%201.0%20release.)

Code snippets which worked (with the deprecated package) are-
In UiProvider i added backstage item-
```
.....
.....
public provideBackstageItems(): BackstageItem[] {
    const backstageItems: BackstageItem[] = [];
    backstageItems.push(
      BackstageItemUtilities.createActionItem(
        "iModelSelector",
        100,
        97,
        async () => {
          const nestedFrontstageProvider = new NestedFrontstageProvider();
          const frontstageDef = nestedFrontstageProvider.initializeDef();
          await FrontstageManager.openNestedFrontstage(frontstageDef);
        },
        "Open Online IModel",
        undefined,
        ""
      )
    );
    return backstageItems;
  }
........
.......
```

In NestedFrontstageProvider.tsx, i have-

```
class IModelSelectorControl extends ContentControl {
    constructor(info: ConfigurableCreateInfo, options: any) {
        super(info, options);
        this.reactNode = <IModelSelector onIModelSelected={this._onSelectIModel} />;
    }

    // called when an imodel has been selected on the IModelSelect
    private _onSelectIModel = async (iModelInfo: IModelInfo) => {
        history.push(`?contextId=${iModelInfo.projectInfo.wsgId}&iModelId=${iModelInfo.wsgId}`);
        window.location.reload()
       };
}
export class NestedFrontstageProvider extends FrontstageProvider {
    private _contentLayoutDef: ContentLayoutDef;
    constructor() {
        super();
        this._contentLayoutDef = new ContentLayoutDef({});
    }
    public get frontstage(): React.ReactElement<FrontstageProps> {
        const contentGroup: ContentGroup = new ContentGroup({
            contents: [
                {
                    classId: IModelSelectorControl,
                },
            ],
        });
        return (
            <Frontstage
                id="NestedFrontstageProvider"
                defaultTool={CoreTools.selectElementCommand}
                defaultLayout={this._contentLayoutDef}
                contentGroup={contentGroup}
                contentManipulationTools={
                    <Zone
                        widgets={[
                            <Widget isFreeform={true} element={<ToolWidget appButton={NestedFrontstage.backToPreviousFrontstageCommand} />} />,
                        ]}
                    />
                }
            />
        );
    }
}
```
What should be the approach to achieve this feature for my app. Is there any other way to get this working without using deprecated package??

Thanks

**Comment by aruniverse** [comment]
Although the @itwin/imodel-browser-react package is versioned 0.x and considered _unstable_, it should be safe enough for you to use, we'll look into upgrading it to 1.x. 

You could also create your own component that does the same thing if you didn't want to use the above pkgs.

**Reply by kmurari103** [reply]
Is there any sample code that can be referred to achieve this using @itwin/imodel-browser-react? 

Thanks

**Reply by aruniverse** [reply]
Have you taken a look at our Storybook example, https://github.com/iTwin/admin-components-react/tree/main/packages/apps/storybook/src/imodel-browser ?
